spec/src/main/asciidoc/configexamples.asciidoc

```diff
@@ -119,3 +119,43 @@ public class InjectedConfigUsageSample {
 }
 ----
 
+=== Aggregate related properties into a single POJO Example
+
+When injecting a number of related configuration properties, it can be tedious to repeat the statement of `ConfigProperty` in scatter places. 
+Since they are related, it makes more sense to aggregate them into a single property class. 
+
+MicroProfile Config provides a way to look up a number of configuration properties starting with the same prefix using the `@ConfigProperties` annotation.
+Considering the following config source:
+
+[source, text]
+----
+config_ordinal = 120
+server.host = localhost
+server.port=9080
+server.endpoint=query
+----
+
+In order to retrieve the above properties in a single property class, you can use the `@ConfigProperties` annotation with the specified prefix.
+All of the fields specified in the class maps to the corresponding configuration properties. The field x maps to the configuration property <prefix>.x.
+
+[source, java]
+----
+
+@ConfigProperties("server")
+public interface ServerDetails {
```

Comment:
We should support both classes and interfaces, right?

Reply:
I am undecided whether to support both. I started with class. For supporting class, we can use public variables. For support interface, we can just use methods (following JavaBean naming convention) .

Reply:
Actually, I changed my mind again. Let me just start with class first and then we can add interface. I don't see the point of supporting both. Supporting class is neater, because in order to support interface, we have to adopt JavaBean naming convention, which adds some level of requirement, where supporting classes is quite straightforward.

Reply:
Ok